utk/utk.go

```diff
@@ -6,35 +6,73 @@ import (
 	"io/ioutil"
 	"log"
 	"os"
+	"context"
 	"encoding/json"
 
 	"github.com/linuxboot/fiano/uefi"
+	"github.com/google/subcommands"
 )
 
-func main() {
-	flag.Parse()
-	if len(flag.Args()) == 0 {
-		log.Fatal("A file name is required")
+// Parse subcommand
+type parseCmd struct {
+}
+
+func (*parseCmd) Name() string {
+	return "parse"
+}
+
+func (*parseCmd) Synopsis() string {
+	return "Parse rom file and print JSON summary to stdout"
+}
+
+func (*parseCmd) Usage() string {
+	return "parse <path-to-rom-file>\n"
+}
+
+func (*parseCmd) SetFlags(_ *flag.FlagSet) {}
+
+func (*parseCmd) Execute(_ context.Context, f *flag.FlagSet, _ ...interface{}) subcommands.ExitStatus {
+	if len(f.Args()) == 0 {
```

Comment:
This is all fine, one thought: I once had it suggested to me that if you use f.Args() more than once, you should just do args := f.Args() since the call cost is nontrivial. I don't necessarily agree with this, but thought I'd mention it.

Reply:
chris, in case you doing me on this, it was a comment from Adin :-)